Source/WebCore/platform/gstreamer/GStreamerQuirks.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2024 Igalia S.L
+ * Copyright (C) 2024 Metrological Group B.V.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * aint with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#if USE(GSTREAMER)
+
+#include "GRefPtrGStreamer.h"
+#include "MediaPlayer.h"
+#include <wtf/Forward.h>
+#include <wtf/RefCounted.h>
+#include <wtf/Vector.h>
+
+namespace WebCore {
+
+enum class ElementRuntimeCharacteristics : uint8_t {
+    IsMediaStream = 1 << 0,
+    HasVideo = 1 << 1,
+    HasAudio = 1 << 2,
+    IsLiveStream = 1 << 3,
+};
+
+class GStreamerQuirkBase {
+    WTF_MAKE_FAST_ALLOCATED;
+
+public:
+    GStreamerQuirkBase() = default;
+    virtual ~GStreamerQuirkBase() = default;
+
+    virtual const char* identifier() = 0;
+};
+
+class GStreamerQuirk : public GStreamerQuirkBase {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    GStreamerQuirk() = default;
+    virtual ~GStreamerQuirk() = default;
+
+    virtual bool isPlatformSupported() const { return true; }
+    virtual GstElement* createWebAudioSink() { return nullptr; }
+    virtual bool configureElement(GstElement*, const OptionSet<ElementRuntimeCharacteristics>&) { return false; }
```

Comment:
Would you mind exposing pipeline configuration via Quirk as well? Currently we maintain a patch to set custom context on the pipeline to signal limited resource usage configuration: https://code.rdkcentral.com/r/plugins/gitiles/rdk/components/generic/rdk-oe/meta-rdk-ext/+/refs/heads/rdk-next/recipes-extended/wpe-webkit/files/2.38/comcast-XIONE-12272-configure-video-resource-usage-w.patch#22, it'd be great to have an option to put it into Westeros quirk.

Reply:
Yeah I can add this code in the Westeros quirk, no problem I think.